cscs-checks/apps/amber/amber_check.py

```diff
@@ -57,65 +54,35 @@
     },
 }
 
+REFERENCE_ENERGY = {
+    # every system has a different reference energy and drift
+    'Cellulose_production_NVE': (-443246.0, 5.0E-05),
+    'FactorIX_production_NVE': (-234188.0, 1.0E-04),
+    'JAC_production_NVE_4fs': (-44810.0, 1.0E-03),
+    'JAC_production_NVE': (-58138.0, 5.0E-04),
+}
 
-class AmberBaseCheck(rfm.RunOnlyRegressionTest):
-    valid_prog_environs = ['builtin']
-    strict_check = False
-    modules = ['Amber']
-    extra_resources = {
-        'switches': {
-            'num_switches': 1
-        }
-    }
-    maintainers = ['VH', 'SO']
-    tags = {'scs', 'external-resources'}
 
+class AmberCheck(AmberBaseCheck):
+    modules = ['Amber']
     benchmark = parameter([
         # NVE simulations
         'Cellulose_production_NVE',
         'FactorIX_production_NVE',
         'JAC_production_NVE_4fs',
         'JAC_production_NVE',
     ])
-
-    @run_after('init')
-    def download_files(self):
-        self.prerun_cmds = [
-            # cannot use wget because it is not installed on eiger
-            f'curl -LJO https://github.com/victorusu/amber_benchmark_suite'
-            f'/raw/main/amber_16_benchmark_suite/PME/{self.benchmark}.tar.bz2',
-            f'tar xf {self.benchmark}.tar.bz2'
-        ]
-
-    @run_after('init')
-    def set_energy_and_tolerance_reference(self):
-        self.ener_ref = {
-            # every system has a different reference energy and drift
-            'Cellulose_production_NVE': (-443246, 5.0E-05),
-            'FactorIX_production_NVE': (-234188, 1.0E-04),
-            'JAC_production_NVE_4fs': (-44810, 1.0E-03),
-            'JAC_production_NVE': (-58138, 5.0E-04),
+    strict_check = False
+    extra_resources = {
+        'switches': {
+            'num_switches': 1
         }
+    }
+    references = REFERENCE_ENERGY
+    output_file = 'amber.out'
+    maintainers = ['VH', 'SO']
 
-    @run_after('setup')
-    def set_executable_opts(self):
-        self.executable_opts = ['-O',
-                                '-i', self.input_file,
-                                '-o', self.output_file]
-        self.keep_files = [self.output_file]
-
-    @run_after('setup')
-    def set_sanity_patterns(self):
-        energy = sn.extractsingle(r' Etot\s+=\s+(?P<energy>\S+)',
-                                  self.output_file, 'energy', float, item=-2)
-        energy_reference = self.ener_ref[self.benchmark][0]
-        energy_diff = sn.abs(energy - energy_reference)
-        ref_ener_diff = sn.abs(self.ener_ref[self.benchmark][0] *
-                               self.ener_ref[self.benchmark][1])
-        self.sanity_patterns = sn.all([
-            sn.assert_found(r'Final Performance Info:', self.output_file),
-            sn.assert_lt(energy_diff, ref_ener_diff)
-        ])
+    run_after('init')(bind(ut.define_reference))
 
     @run_after('setup')
```

Comment:
You can merge this hook with the one below and make it `run_before('performance')`